src/rewriter/README.md

```diff
@@ -94,17 +112,26 @@ processing are controlled by `rewriter.py` and `parser.py`.
 The symbols usable in `(<id> <expr>+)` can be found in `node.py`. The sorts
 available can be found in `mkrewrites.py`.
 
-
 ## Changes from Previous Works
 
 In comparison to
 [fmcad22](https://ieeexplore.ieee.org/abstract/document/10026573), we have
-removed the `let` and `cond` expressions in favour of more atomic rewrite rules.
-`let` is replaced by `def` which allow symbols to be shared across the
-condition, source, and target terms.
-
-We also deleted the `const` modifier since the evaluation of constant
-expressions should be handled elsewhere.
+removed the `let` expression.  `let` is replaced by `def` which allow symbols to
+be shared across the condition, source, and target terms.
+
+We also deleted the `const` modifier the rationale is that there are 3 reasons to use `:const`:
+1. There should not be a rule that is only sound when its arguments are constant, so `:const` should not exist in this case.
+2. The evaluation of operator applications of all constant arguments is handled by the rewriter, and therefore rules evaluating constant expressions should not exist in RARE
+3. There are some conditional rules in `theory/bv` which only apply when the
+   arguments satisfy a condition. In this case, we could match the required argument using a `(bv v n)` term with variable bit width `n` instead of an abstract bitvector term `(y ?BitVec)`. For example,
+``` lisp
+(define-cond-rule bv-udiv-pow2-1p
+  ((x ?BitVec) (v Int) (n Int))
+  (def (power (int.log2 v)))
+  (and (int.ispow2 v) (> v 1))
+  (bvudiv x (bv v n))
+  (concat (bv 0 power) (extract (- n 1) power x)))
+```
 
 More information can be found in other works but beware of changes.
```

Comment:
I think this might be to negative also it is quite inconcrete. Maybe rather have related work section or refer to the changes section you have above.

Reply:
This sentence "More information can be found in other works but beware of changes." was the result of one of our one to one meetings. Do you want it deleted?

Reply:
No not necessarily, it is good to help people find more information and make them aware that there might be changes. But it would be good to be more concrete where they can find more information (you already mention the fmcad paper at the beginning of that section but other sounds like you mean work different from that?). Maybe this is controversial, I'll let the other's chime in.

Reply:
sounds good. I'll wait for others inputs on this one